tanner/new_server.py

```diff
@@ -0,0 +1,108 @@
+import asyncio
+import json
+import logging
+
+import uvloop
+import yarl
+from aiohttp import web
+
+from tanner import api, dorks_manager, session_manager, redis_client
+from tanner.config import TannerConfig
+from tanner.emulators import base
+from tanner.reporting.log_local import Reporting as local_report
+from tanner.reporting.log_mongodb import Reporting as mongo_report
+
+asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
+
+
+class TannerServer:
+    def __init__(self):
+        base_dir = TannerConfig.get('EMULATORS', 'root_dir')
+        db_name = TannerConfig.get('SQLI', 'db_name')
+
+        self.session_manager = session_manager.SessionManager()
+        self.dorks = dorks_manager.DorksManager()
+        self.api = api.Api()
+        self.base_handler = base.BaseHandler(base_dir, db_name)
+        self.logger = logging.getLogger(__name__)
+        self.redis_client = None
+
+    @staticmethod
+    def _make_response(msg):
+        response_message = dict(
+            version=1,
+            response=dict(message=msg)
+        )
+        return response_message
+
+    @staticmethod
+    async def default_handler(request):
+        return web.Response(text="Tanner server")
+
+    async def handle_event(self, request):
+        data = await request.read()
+        try:
+            data = json.loads(data.decode('utf-8'))
+            path = yarl.unquote(data['path'])
+        except (TypeError, ValueError, KeyError) as error:
+            self.logger.error('error parsing request: %s', data)
+            response_msg = self._make_response(msg=type(error).__name__)
+        else:
+            session = await self.session_manager.add_or_update_session(
+                data, self.redis_client
+            )
+            self.logger.info('Requested path %s', path)
+            await self.dorks.extract_path(path, self.redis_client)
+            detection = await self.base_handler.handle(data, session, path)
```

Comment:
Maybe rename `handle()` into `detect` or `classify`? Should probably go into a separate PR :)